sorted_db/db.go

```diff
@@ -186,6 +204,42 @@ func (db *DB) Search(needle []byte) []byte {
 	return nil
 }
 
+// RangeMatch uses binary searches to look for startNeedle and (if not nil)
+// endNeedle. Returns all full match lines that fall between startNeedle and
+// endNeedle, inclusive. startNeedle and endNeedle should already both end
+// with the record separator.
+func (db *DB) RangeMatch(startNeedle []byte, endNeedle []byte) []byte {
+	db.RLock()
+
+	if db.size <= 0 {
+		panic("DB not Mapped")
+	}
+	startRecord := db.findFirstRecordAfterMatch(startNeedle, true)
+	if startRecord < 0 || startRecord == db.size {
+		db.RUnlock()
+		return nil
+	}
+	previous := lastIndexByte(db.data, startRecord, db.LineEnding)
+	// eat the line ending or (if there is no line ending) move to the start
+	startIndex := previous + 1
+
+	endIndex := db.size
+	if endNeedle != nil {
+		endRecord := db.findFirstRecordAfterMatch(endNeedle, false)
+		if endRecord >= 0 && endRecord < db.size {
+			previous := lastIndexByte(db.data, endRecord, db.LineEnding)
+			// eat the line ending or (if there is no line ending) move to the start
+			endIndex = previous + 1
+		}
+	}
+
+	// intentionally make a copy of data
+	records := []byte(db.data[startIndex:endIndex])
```

Comment:
Crushed that bug.

![root-beer-crushes-it](https://cloud.githubusercontent.com/assets/73363/9238425/fdf98d10-4106-11e5-9f48-1d40dca9b8fd.gif)